### PR TITLE
Adds OSS::SplitModal component & flaky test fixing

### PR DESCRIPTION
**Pull request by Miexil**
### What does this PR do?
Adds OSS::SplitModal component & flaky test fixing

Related to: #https://linear.app/upfluence/issue/ENG-116/[plg-prep]-move-frankenmodal-component-to-oss

### What are the observable changes?
New OSS::SplitModal component added:
- Common BaseModal created (to share code between SplitModal & ModalDialog)
- Tests added
- Documentation added
- Flaky test fixing 

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: app/styles/split-modal.less [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    max-width: 1050px;
+    min-height: 550px;
+  }
+
+  &__content {
+    border-right: 1px solid var(--color-gray-200);
+
+    footer {
+      flex: 0;
+      padding: @spacing-xx-sm @spacing-x-sm;
```
Comment (JulienVannier66) [inline app/styles/split-modal.less:13]:
```suggestion
      padding: var(--spacing-px-12) var(--spacing-px-18);
```

File: tests/integration/components/o-s-s/modal-dialog-test.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { render, setupOnerror, triggerKeyEvent } from '@ember/test-helpers';
 import hbs from 'htmlbars-inline-precompile';
 import sinon from 'sinon';
 import click from '@ember/test-helpers/dom/click';
+import { resetOnerror } from '@ember/test-helpers/setup-onerror';
```
Comment (JulienVannier66) [inline tests/integration/components/o-s-s/modal-dialog-test.ts:7]: Keep that?

File: tests/integration/components/o-s-s/split-modal-test.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, triggerKeyEvent, click, resetOnerror } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+import setupOnerror from '@ember/test-helpers/setup-onerror';
```
Comment (JulienVannier66) [inline tests/integration/components/o-s-s/split-modal-test.ts:6]: Another one

**Review by phndiaye — APPROVED** [review]

**Review by JulienVannier66 — APPROVED** [review]
Otherwise LGTM

**Review by aprentout — APPROVED** [review]

**Pull request merged**